modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/authbst/impl/BSTProviderImpl.java

```diff
@@ -54,30 +56,22 @@
     public Response getBST(Credentials credentials) throws WindowsDeviceEnrolmentException {
 
         String domainUser = credentials.getUsername();
-        String domain = "";
-        String password = credentials.getPassword();
-
+        String userToken = credentials.getUsertoken();
         try {
-            if (authenticate(domainUser, password, domain)) {
-                String challengetoken = DeviceUtil.generateRandomToken();
-                Token tokenbean = new Token();
-                tokenbean.setChallengeToken(challengetoken);
-                DeviceUtil.persistChallengeToken(tokenbean.getChallengeToken(), "", domainUser);
-
-                return Response.ok().entity(tokenbean.getChallengeToken()).build();
-            } else {
-                String msg = "Authentication failure due to incorrect credentials.";
-                log.error(msg);
-                return Response.status(403).entity("Authentication failure").build();
-            }
-        } catch (AuthenticationException e) {
-            String msg = "Failure occurred in user authentication process.";
-            log.error(msg);
-            throw new WindowsDeviceEnrolmentException(msg);
+            Token tokenBean = new Token();
+            tokenBean.setChallengeToken(userToken);
+            DeviceUtil.persistChallengeToken(tokenBean.getChallengeToken(), "", domainUser);
+            JSONObject tokenContent = new JSONObject();
+            tokenContent.put("UserToken", userToken);
+            return Response.ok().entity(tokenContent.toString()).build();
         } catch (DeviceManagementException e) {
             String msg = "Failure occurred in generating challenge token.";
             log.error(msg);
             throw new WindowsDeviceEnrolmentException(msg);
+        } catch (JSONException e) {
+            String msg = "Failure occurred in generating challenge token Json.";
+            log.error(msg);
```

Comment:
I believe, it's okay to use "log and throw" in this particular instance as "BSTProvider" is what gets exposed externally. Therefore, it makes sense to log the errors before the exceptions, etc are dispatched out of the service.